llvm/test/MC/AArch64/arm64ec-disallowed-regs.s

```diff
@@ -0,0 +1,75 @@
+// RUN: llvm-mc -triple arm64ec-pc-windows-msvc < %s 2> %t.log
+// RUN: FileCheck %s --check-prefix=CHECK-ERR < %t.log
+
+
+// ---- disallowed x registers ----
+orr x13, x0, x1             // x13
+// CHECK-ERR: warning: this instruction uses disallowed registers.
+orr x14, x2, x3             // x14
+// CHECK-ERR: warning: this instruction uses disallowed registers.
+orr x4, x23, x5             // x23
+// CHECK-ERR: warning: this instruction uses disallowed registers.
+orr x6, x7, x24             // x24
+// CHECK-ERR: warning: this instruction uses disallowed registers.
+orr x28, x8, x9             // x28
+// CHECK-ERR: warning: this instruction uses disallowed registers.
```

Comment:
With a test sequence like this, it's kinda hard to keep track of which instruction maps to which error. I was kind of wanting to have some separate instruction here which would not trigger the warning as well, but it's not very easy to verify that. Or perhaps we should have a separate test file which only uses allowed registers, and validate that it produces no warnings? (And also maybe as another test aspect - test that this file produces no warnings when assembled as regular `aarch64-windows-msvc`.)

Reply:
The idea for this testcase was that each line uses one disallowed register and generates a warning. I can include another testcase that uses allowed registers only (and I'll also check for no warnings when assembled for regular AArch64 on this testcase).

Reply:
+1 to building this as `aarch64-windows-msvc` and validating there are no warnings.
As for making sure that we can build other assembly as Arm64EC, I wouldn't add a dedicated test case, but instead add Arm64EC as a run in some other tests for AArch64 that already exist.

Reply:
Thanks, now as the warning message includes the name of the forbidden register, it's much less of a blind sequence (where testing quality depends on the number of warnings exactly matching the number of instructions) - now this is less of a concern to me.